pkg/operator/staticpod/controller/revision/revision_controller.go

```diff
@@ -205,6 +205,12 @@ func (c RevisionController) createNewRevision(revision int32) error {
 	}}
 
 	for _, cm := range c.configMaps {
+		// If the target config map was already created, do not remove it as there might be installer pods in flight and we don't
+		// want the installer pod to fail. The installer should finish installing the revision and if the source configmap changed
+		// the new revision will replace the old revision afterwards.
+		if _, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(nameFor(cm.Name, revision), metav1.GetOptions{}); err == nil {
```

Comment:
IMO this is wrong: it leaves old data and does not deployment of new data.

Reply:
new data should trigger new revisions, not update existing "in-flight" revisions and cause the installer pods to fail?

Reply:
```
$ cat operator.log | grep 'RevisionCreate' | grep -o 'Revision .* created'
Revision 1 created
Revision 2 created
Revision 3 created
Revision 4 created
Revision 5 created
Revision 6 created
Revision 7 created
```

It looks like, we wrote all 7 revisions in order, however we triggered revision 2, 3 and 4 twice once with existing configmap (the one installer pod that ran successfully and second time because the `isLatestRevisionCurrent` found the configmap was deleted... Which is very odd because `LatestAvailableRevision` is being wrote when we send `RevisionCreate` event...

Status update has built in retry on conflict mechanism, so it is unlikely we fail to update so many times we have to retry the reconciliation. However, in that case, I would probably skip the "already created" resources for the given revision?